src/mrpro/data/SpatialDimension.py

```diff
@@ -109,6 +108,16 @@ def from_array_zyx(
 
         return SpatialDimension(z, y, x)
 
+    def apply_(self, function: Callable[[T], T] | None = None, **_) -> Self:
```

Comment:
why do we need `**_`? For later overloads?

Reply:
it overloads the one from MoveDataMixin, which has additional keyword only parameters (recurse and memo), which Ithink don#t make sense here. But we could also have them and ignore them..
The only reason for apply_ here is to define it with a function Callable[[T], T], as the function does not have to deal with any other children-types.